Add (Go) Slog documentation (#10976)
getsentry/sentry-docs: docs/platforms/go/guides/slog/index.mdx

---
title: Slog
description: "Slog is a structured logging library for Go, introduced in Go 1.21. This guide demonstrates how to integrate slog with Sentry to capture and send logs to Sentry."
---

> For a quick reference, there is a [complete example](https://github.com/getsentry/sentry-go/tree/master/_examples/slog) at the Go SDK source code repository.

Reviewer: This link currently gives me a 404

Author: There is an open PR in the Go-SDK repository (linked in PR description). Once it's merged, the link should work.

> [Go Dev-style API documentation](https://pkg.go.dev/github.com/getsentry/sentry-go/slog) is also available.

Reviewer: This link also doesn't work for me. I'm also wondering if it's necessary to include since it seems to be more of a reference? Can we just leave it out or maybe move to the bottom of the page if it's really a helpful reference?

Reviewer: ```suggestion
Go API documentation for the [`sentryslog` package](https://pkg.go.dev/github.com/getsentry/sentry-go/slog) is also available.
```
After looking at your other PR for logrus, I think adjusting the wording here to something like this makes the relevance of the link clearer. Feel free to adjust as needed!
